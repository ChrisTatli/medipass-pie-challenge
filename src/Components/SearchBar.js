import React from 'react'
import TextField from '@material-ui/core/TextField';
import Button from '@material-ui/core/Button';

const SearchBar = (props)=>{
    return(
        <div>
            <TextField style={{padding: 24}}
                        id="searchInput"
                        placeholder="Search for Pies"
                        margin="normal"
                        onChange={props.onSearchChange}
            />
            {props.order === "desc" ? (
            <Button 
              size="small" 
              onClick ={props.changeOrder}
              color = "primary"
            >
              Ascending Price
            </Button>
            ):(
            <Button 
              size="small" 
              onClick ={props.changeOrder}
              color = "primary"
            >
              Descending Price
            </Button>
            )}
        </div>
    )
}


export default SearchBar