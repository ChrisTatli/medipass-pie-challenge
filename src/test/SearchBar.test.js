import React from 'react';
import SearchBar from '../Components/SearchBar';
import renderer from 'react-test-renderer';


test('SearchBar component renders a snapshot', () =>{
    const order = "asc"
    const component = renderer.create(
        <SearchBar 
            order={order}
        />
    );

    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();



});