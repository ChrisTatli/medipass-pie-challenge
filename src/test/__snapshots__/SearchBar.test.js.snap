// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchBar component renders a snapshot 1`] = `
<div>
  <div
    aria-describedby={undefined}
    className="MuiFormControl-root-1 MuiFormControl-marginNormal-2"
    style={
      Object {
        "padding": 24,
      }
    }
  >
    <div
      className="MuiInput-root-5 MuiInput-formControl-6 MuiInput-underline-9"
    >
      <input
        aria-invalid={false}
        autoComplete={undefined}
        autoFocus={undefined}
        className="MuiInput-input-13"
        defaultValue={undefined}
        disabled={false}
        id="searchInput"
        name={undefined}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={undefined}
        onKeyUp={undefined}
        placeholder="Search for Pies"
        readOnly={undefined}
        required={false}
        rows={undefined}
        type="text"
        value={undefined}
      />
    </div>
  </div>
  <button
    className="MuiButtonBase-root-42 MuiButton-root-18 MuiButton-text-20 MuiButton-textPrimary-21 MuiButton-flat-23 MuiButton-flatPrimary-24 MuiButton-sizeSmall-39"
    disabled={false}
    onBlur={[Function]}
    onClick={undefined}
    onFocus={[Function]}
    onKeyDown={[Function]}
    onKeyUp={[Function]}
    onMouseDown={[Function]}
    onMouseLeave={[Function]}
    onMouseUp={[Function]}
    onTouchEnd={[Function]}
    onTouchMove={[Function]}
    onTouchStart={[Function]}
    tabIndex="0"
    type="button"
  >
    <span
      className="MuiButton-label-19"
    >
      Descending Price
    </span>
    <span
      className="MuiTouchRipple-root-45"
    />
  </button>
</div>
`;
